tests/test_hashring.py

```diff
@@ -14,24 +14,21 @@ def __repr__(self):
         return f"<Node {self.id}>"
 
 
-class HashRingTest(unittest.TestCase):
-    def setUp(self):
-        self.node0 = Node(0)
-        self.node1 = Node(1)
-        self.node2 = Node(2)
+@pytest.fixture
+def hash_ring():
+    return HashRing([Node(i) for i in range(3)])
 
-        self.nodes = [self.node0, self.node1, self.node2]
-        self.ring = HashRing(self.nodes)
```

Comment:
Unfortunately the diff here is not great, but `setUp()` code can generally be converted to a `pytest.fixture`. They can either return a value or yield a value in order to perform cleanup after a test finishes (to act as a `tearDown()`).

For more information: https://docs.pytest.org/en/stable/fixture.html#quick-example